src/AppInstallerCLICore/Workflows/ShellExecuteInstallerHandler.cpp

```diff
@@ -65,7 +65,20 @@ namespace AppInstaller::CLI::Workflow
             const std::map<ManifestInstaller::InstallerSwitchType, Utility::NormalizedString>& installerSwitches = context.Get<Execution::Data::Installer>()->Switches;
 
             // Construct install experience arg.
-            if (context.Args.Contains(Execution::Args::Type::Silent) && installerSwitches.find(ManifestInstaller::InstallerSwitchType::Silent) != installerSwitches.end())
+            auto installerType = context.Get<Execution::Data::Installer>()->InstallerType;
+            if (context.Args.Contains(Execution::Args::Type::Silent) && installerType == ManifestInstaller::InstallerTypeEnum::Nullsoft)
+            {
+                installerArgs += "/S";

```

Comment:
>installerArgs += "/S"; [](start = 16, length = 22)

Thanks for your input.

In current implementation, the default args for known installer types are stored in
ManifestInstaller::GetDefaultKnownSwitches as a map in [https://github.com/microsoft/winget-cli/blob/master/src/AppInstallerRepositoryCore/Manifest/ManifestInstaller.cpp](https://github.com/microsoft/winget-cli/blob/master/src/AppInstallerRepositoryCore/Manifest/ManifestInstaller.cpp)

The default args are auto populated during manifest parsing if the original manifest does not declare the args.

Please update there if you feel the default args in our map needs improvement.

Thanks.